spipe/main.c

```diff
@@ -176,16 +179,23 @@ main(int argc, char * argv[])
 	}
 
 	/* Loop until we die. */
-	do {
-		if (events_run()) {
-			warnp("Error running event loop");
-			exit(1);
-		}
-	} while (1);
+	if (events_spin(&conndone)) {
+		warnp("Error running event loop");
+		exit(1);
```

Comment:
I do want it to be `goto errN` eventually, but I haven't got a good way of cleaning up after `pushbits`.  I'm implementing the error path in stages (as I did with `spiped`), so I'd prefer to get this merged, then I'll take another look at my various branches that attacked `pushbits` and discuss it with you tomorrow morning.